appinfo.go

```diff
@@ -20,9 +30,55 @@ type AppengineInfo interface {
 	Zone() string
 }
 
+var (
+	zone    string
+	zoneMtx sync.Mutex
+)
+
+func Zone() string {
+	zoneMtx.Lock()
+	defer zoneMtx.Unlock()
+
+	if zone == "" {
+		z, err := metadata.Zone()
+		if err != nil {
+			panic(err)
+		}
+		zone = z
+	}
+
+	return zone
+}
+
 var (
 	IsDevAppServer = false
 	IsFlex         = appengine.IsFlex
 	IsSecondGen    = appengine.IsSecondGen
 	IsStandard     = appengine.IsStandard
 )
+
+// Don't call this.  It exists to make NewAppengineInfoFromContext mockable
+func InternalNewAppengineInfoFromContext(c context.Context) AppengineInfo {
+	// if running in K8s, the following environment variable will always be set
+	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" {
+		config, err := rest.InClusterConfig()
```

Comment:
we'll need to be careful with how many times we call this I'm guessing.  Might need to cache the response for some amount of time.

Reply:
Luckily, it isn't a network call. It is just information gathered and present by default on the container's filesystem.

Reply:
Others below, also do not make network calls